### DISCUSSION TITLE
text to speech error

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#922] [Q&A] [answered]
#include "SimpleTTS.h"

After adding above header  I am getting below error

![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/8c572e09-d891-40bd-b3d8-71bcac8e7f27)

**Comment by pschatzmann** [comment, marked as answer]
I can't reproduce your issue.
You must have some other issue: just including some header file can't have such an impact...

**Reply by aqildad-create** [reply]
as per below threat i think its because RAM over flow .

![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/84836277-f4c9-4294-9766-da03a7c8a3fe)

**Comment by aqildad-create** [comment]
Is their any way to use your API and store the file in SD card.

**Comment by pschatzmann** [comment]
Unlikely: it does not use much:
Sketch uses 740438 bytes (23%) of program storage space. Maximum is 3145728 bytes.
Global variables use 23632 bytes (7%) of dynamic memory, leaving 304048 bytes for 

If you check the [Dictionary implementations](https://pschatzmann.github.io/arduino-simple-tts/docs/html/classsimple__tts_1_1_audio_dictionary_base.html) you can see that SD is one of the options

All the necessary information is available in the [Wiki](https://github.com/pschatzmann/arduino-simple-tts/wiki)!

**Comment by aqildad-create** [comment]
#include "SimpleTTS.h"
#define USE_SD
#include "AudioDictionarySD.h"
#define  PIN_CS 5
AudioDictionarySD dictionary("/tts","mp3", PIN_CS);


now after using above i am getting below errors 

![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/f4594d55-e13b-4c3f-a189-8b42eed9d84f)

**Comment by pschatzmann** [comment]
I have just committed some changes to this library to reflect the updates in the AudioTools.
It should compile now again.

**Comment by aqildad-create** [comment]
#include "SimpleTTS.h"
#define USE_SD
#include "AudioDictionarySD.h"
#define  PIN_CS 5
AudioDictionarySD dictionary("/tts","mp3", PIN_CS);


After using above line and update the API it compile but getting exception error . 

before adding tts file space details are below .

Sketch uses 932689 bytes (47%) of program storage space. Maximum is 1966080 bytes.
Global variables use 51096 bytes (15%) of dynamic memory, leaving 276584 bytes for local variables. Maximum is 327680 bytes.
esptool.py v4.5.1

After tts header space details are below 
Invalid library found in C:\Users\hp\Documents\Arduino\libraries\downloads: no headers files (.h) found in C:\Users\hp\Documents\Arduino\libraries\downloads
Sketch uses 1237885 bytes (62%) of program storage space. Maximum is 1966080 bytes.
Global variables use 52200 bytes (15%) of dynamic memory, leaving 275480 bytes for local variables. Maximum is 327680 bytes.


after huge APP below is storage detail .so  problem is some were else which  i cant figure out . 
Sketch uses 1237885 bytes (39%) of program storage space. Maximum is 3145728 bytes.
Global variables use 52200 bytes (15%) of dynamic memory, leaving 275480 bytes for local variables. Maximum is 327680 bytes.
esptool.py v4.5.1
Serial port COM12



This is the exception  I am getting  
 
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/3d8fe0f0-a85f-4736-90ef-6901d47bcc09)

**Reply by pschatzmann** [reply]
I never used an S2. 
Did you try with a regular ESP32

**Comment by pschatzmann** [comment]
https://github.com/espressif/esp-idf/issues/5503

**Comment by aqildad-create** [comment]
Its esp32 wroom

**Comment by pschatzmann** [comment]
Check your stacktrace: You are using esp32/esp32S2
That would explain why it's not working...

**Comment by aqildad-create** [comment]
yes but i select in board esp32 wroom da module . and board i have is esp32 wroom-32 .. I don't now why  i am getting this exception .pls guide 


![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/98649a5f-d3e6-4b21-9f21-0d7fc29f6b60)


![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/3edf3be4-f058-4af3-9d54-b2dca9dc3153)


this is the board physically connected

![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/d407a66b-f69d-4155-82aa-50e09543037a)

**Reply by pschatzmann** [reply]
Strange: did you try to upload by selecting the regular ESP32 Dev Module ?
Try to activate the Core Debug Level, maybe this gives you some additional info
I would also set erase the flash before upload

**Comment by aqildad-create** [comment]
now  select board esp32 dev module . . which is regular esp32 dev board right ? 
and core debug level to info 
and erase all sketch before code upload to enable 
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/f8b5af0b-4121-489b-b01c-54bf71a58968)

now physical board connoted is esp32   ai thnker 
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/769dff97-a4ed-412f-95d0-85e54efeafd5)

and i am getting below exception 
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/9c833b4c-bb35-4bdc-8a87-46e7265070ab)

this is complete seral out error


ELF file SHA256: 762362c49314f16b

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13924
ho 0 tail 12 room 4
load:0x40080400,len:3600
entry 0x400805f0

assert failed: esp_startup_start_app_common port_common.c:81 (res == pdTRUE)


Backtrace: 0x40083cbd:0x3ffe3a50 0x4008d44d:0x3ffe3a70 0x40092f59:0x3ffe3a90 0x40090949:0x3ffe3bc0 0x4017993a:0x3ffe3bf0 0x400f79d7:0x3ffe3c10 0x400833c2:0x3ffe3c40 0x40079276:0x3ffe3c90 |<-CORRUPTED




ELF file SHA256: 762362c49314f16b

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13924
ho 0 tail 12 room 4
load:0x40080400,len:3600
entry 0x400805f0

assert failed: esp_startup_start_app_common port_common.c:81 (res == pdTRUE)


Backtrace: 0x40083cbd:0x3ffe3a50 0x4008d44d:0x3ffe3a70 0x40092f59:0x3ffe3a90 0x40090949:0x3ffe3bc0 0x4017993a:0x3ffe3bf0 0x400f79d7:0x3ffe3c10 0x400833c2:0x3ffe3c40 0x40079276:0x3ffe3c90 |<-CORRUPTED




ELF file SHA256: 762362c49314f16b

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13924
ho 0 tail 12 room 4
load:0x40080400,len:3600
entry 0x400805f0

assert failed: esp_startup_start_app_common port_common.c:81 (res == pdTRUE)


Backtrace: 0x40083cbd:0x3ffe3a50 0x4008d44d:0x3ffe3a70 0x40092f59:0x3ffe3a90 0x40090949:0x3ffe3bc0 0x4017993a:0x3ffe3bf0 0x400f79d7:0x3ffe3c10 0x400833c2:0x3ffe3c40 0x40079276:0x3ffe3c90 |<-CORRUPTED




ELF file SHA256: 762362c49314f16b

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13924
ho 0 tail 12 room 4
load:0x40080400,len:3600
entry 0x400805f0

assert failed: esp_startup_start_app_common port_common.c:81 (res == pdTRUE)


Backtrace: 0x40083cbd:0x3ffe3a50 0x4008d44d:0x3ffe3a70 0x40092f59:0x3ffe3a90 0x40090949:0x3ffe3bc0 0x4017993a:0x3ffe3bf0 0x400f79d7:0x3ffe3c10 0x400833c2:0x3ffe3c40 0x40079276:0x3ffe3c90 |<-CORRUPTED

**Comment by pschatzmann** [comment]
What exectly are you doing ?

I just tried the audio-i2s sketch and it is loading and running perfectly on a ESP32 DevKIT V1. As usual I was selecting the Partition Scheme Hugh APP. But selecting another one was working as well. I tested with  the Arduino  ESP version 2.0.9

**Comment by aqildad-create** [comment]
#include "SimpleTTS.h"
#define USE_SD
#include "AudioDictionarySD.h"
#define PIN_CS 5
AudioDictionarySD dictionary("/tts","mp3", PIN_CS);

yes now   I confirm you that above line or tts speech only work if  i remove below WiFi mode from setup . other wise it is throwing exception. 
why is this and how to avoid this ? 

 WiFi.mode(WIFI_STA);

**Comment by pschatzmann** [comment]
Not sure since I never used this: Maybe this is just using too much RAM...
You could try to decrease the I2S buffers...

**Reply by aqildad-create** [reply]
I need to use WiFi.mode(WIFI_STA); also .

together it is not working .


I think its not related to RAM . some other conflict but i cant trouble shoot further but i need wifi mode also . 

Sketch uses 789765 bytes (40%) of program storage space. Maximum is 1966080 bytes.
Global variables use 33260 bytes (10%) of dynamic memory, leaving 294420 bytes for local variables. Maximum is 327680 bytes.

**Reply by pschatzmann** [reply]
You would really need to query the RAM after opening the WIFI connection. This allocates a lot of RAM dynamically, so you don't see this in your statistics. I did not have any issues with my time-to-speech-url.ino sketch which was using WIFI as well...
If you run out of RAM you can try to use a processor with PSRAM instead...

**Reply by aqildad-create** [reply]
yeah agree separately tts is working with WiFi .

**Reply by pschatzmann** [reply]
MP3 decoding is also using a lot of RAM, so you could use WAV files instead...

**Reply by aqildad-create** [reply]
I can't because WAVE file size is large take lot of time while sending via espnow .

**Comment by aqildad-create** [comment]
HOW to query the RAM after opening the WIFI connection ??

**Reply by pschatzmann** [reply]
https://forum.arduino.cc/t/how-to-compute-the-total-size-of-ram-and-the-usage-size-for-esp32/929955/2
